keps/sig-node/4381-dra-semantic-parameters/README.md

```diff
@@ -0,0 +1,1477 @@
+particular sub-partition to be usable. The initial implementation will use a
+very simplistic approach for satisfying requests: it iterates over the requests
+in order and uses the first sub-partition in a depth-first search which
+matches. This locks that instance into a certain partitioning, which then
+influences whether the next request still can be satisfied. With that in mind,
+a better way to list sub-partitions would be to list the small ones first in
+the NodeResourceSlice, to avoid using a whole GPU for a claim that could have
+used a smaller sub-partition instead.
+
+Eventually, this implementation will have to become smarter. It needs to
+support backtracking and scoring of possible solutions to find the best one
+which satisfies the requests. This could be extended to support finding the
+best solution for a set of pods and their claims.
+
+Another future extension is to express constraints that must be satisfied for
+the selected instances. For example, selecting two cards which are on the same
+PCI root complex may be needed to get the required performance.
+
+Instead of defining a vendor CRD, DRA driver authors or administrators may also
+decide to allow users to create and reference ResourceClaimParameters directly
+in their ResourceClaim. Then the translation step from vendor CRD is not
+needed. The downside is a lack of validation of these user-created
+ResourceClaimParameters.
+
+Resource class parameters are supported the same way. To ensure that
+permissions can be limited to administrators, there's a separate cluster-scoped
+ResourceClassParameters type. Instead of individual requests, one additional
+selector can be specified there which then also must be true for all individual
+requests made with that class:
+
+```yaml
+kind: ResourceClassParameters
+apiVersion: resource.k8s.io/v1alpha2
+
+metadata:
+  name: someArbitraryName
+
+generatedFrom:
+  name: gpu-parameters
+  kind: CardClassParameters
+  apiGroup: dra.example.com
+  uid: foobar-uid
+
+vendorParameters:
+  ...
+
+filters:
+- driverName: cards.dra.example.com
+  partitionable:
+    selector: |-
+      quantities["memory"] <= "16Gi"
+```
+
+In this example, the additional selector expression limits users of this class
+to smaller sub-partitions. Together with limiting the number of claims that
+users are allowed to create for this class (see resource quotas in the core
+KEP) this can ensure that users do not consume too much resources. Allowing
+resource quotas that are based on resource attributes may be a useful future
+enhancement.
+
+### Communicating allocation to the DRA driver
+
+The scheduler decides which resources to use for a claim and how much of
+them. It also needs to pass through the opaque vendor parameters, if there are
+any. All of this information gets stored in the allocation result inside the
+ResourceClaim status. For the example above, the result produced by the
+scheduler is simply the list of IDs of the selected sub-partitions:
```

Comment:
You may want to mention here that this information is copied "as-was" at the time of allocation and elaborate on that (unless you do that later).

Reply:
I agree, I think it's useful to include that here (even if it's reiterated later)